rotkehlchen/chain/evm/decoding/curve/decoder.py

```diff
@@ -505,161 +555,61 @@ def _decode_gauge_events(self, context: DecoderContext) -> DecodingOutput:
         gauge_address = context.tx_log.address
         raw_amount = hex_or_bytes_to_int(context.tx_log.data)
         found_event_modifying_balances = False
+        try:
+            gauge_token = EvmToken(evm_address_to_identifier(
+                address=context.tx_log.address,
+                chain_id=self.evm_inquirer.chain_id,
+                token_type=EvmTokenKind.ERC20,
+            ))
+        except (WrongAssetType, UnknownAsset):
+            # if there is no token, that means the gauge is an old one, without any token
+            gauge_token = None
+
         for event in context.decoded_events:
-            crypto_asset = event.asset.resolve_to_crypto_asset()
             if (
                 event.location_label == provider and
                 event.address == gauge_address and
-                event.balance.amount == asset_normalized_value(amount=raw_amount, asset=crypto_asset)  # noqa: E501
+                event.asset.is_evm_token() and
+                event.balance.amount == asset_normalized_value(
+                    amount=raw_amount,
+                    asset=(evm_asset := event.asset.resolve_to_evm_token()),
+                )
             ):
                 event.counterparty = CPT_CURVE
                 event.product = EvmProduct.GAUGE
                 found_event_modifying_balances = True
                 if context.tx_log.topics[0] == GAUGE_DEPOSIT:
                     event.event_type = HistoryEventType.DEPOSIT
                     event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
-                    event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} into {gauge_address} curve gauge'  # noqa: E501
+                    event.notes = f'Deposit {event.balance.amount} {evm_asset.symbol} into {gauge_address} curve gauge'  # noqa: E501
+                    from_event_type = HistoryEventType.RECEIVE
+                    pair_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    pair_note = 'Receive {{amount}} {symbol} after depositing in {address} curve gauge'  # noqa: E501
```

Comment:
why double {} here for amount?.

edit: Okay I see. Hmm this is weird. Think if you can make it a bit better / easier to read.

At least do add a comment on why the double braces here and below (I know why but need to explain)